packaging/installer/kickstart.sh

```diff
@@ -632,71 +632,78 @@ safe_sha256sum() {
       SYSTYPE="Linux"
 
-      os_release_file=
-      if [ -s "/etc/os-release" ] && [ -r "/etc/os-release" ]; then
-        os_release_file="/etc/os-release"
-      elif [ -s "/usr/lib/os-release" ] && [ -r "/usr/lib/os-release" ]; then
-        os_release_file="/usr/lib/os-release"
-      else
-        warning "Cannot find usable OS release information. Native packages will not be available for this install."
-      fi
-
-      if [ -n "${os_release_file}" ]; then
-        # shellcheck disable=SC1090
-        . "${os_release_file}"
-
-        DISTRO="${ID}"
-        SYSVERSION="${VERSION_ID}"
-        SYSCODENAME="${VERSION_CODENAME}"
-        SYSARCH="$(uname -m)"
+      if [ -z "${SKIP_DISTRO_DETECTION}" ]; then
+        os_release_file=
+        if [ -s "/etc/os-release" ] && [ -r "/etc/os-release" ]; then
+          os_release_file="/etc/os-release"
+        elif [ -s "/usr/lib/os-release" ] && [ -r "/usr/lib/os-release" ]; then
+          os_release_file="/usr/lib/os-release"
+        else
+          warning "Cannot find usable OS release information. Native packages will not be available for this install."
+        fi
 
-        supported_compat_names="debian ubuntu centos fedora opensuse ol arch"
+        if [ -n "${os_release_file}" ]; then
+          # shellcheck disable=SC1090
+          . "${os_release_file}"
 
-        if str_in_list "${DISTRO}" "${supported_compat_names}"; then
-            DISTRO_COMPAT_NAME="${DISTRO}"
+          DISTRO="${ID}"
+          SYSVERSION="${VERSION_ID}"
+          SYSCODENAME="${VERSION_CODENAME}"
         else
-            case "${DISTRO}" in
-            opensuse-leap)
-                DISTRO_COMPAT_NAME="opensuse"
-                ;;
-            cloudlinux|almalinux|rocky|rhel)
-                DISTRO_COMPAT_NAME="centos"
-                ;;
-            artix|manjaro|obarun)
-                DISTRO_COMPAT_NAME="arch"
-                ;;
-            *)
-                DISTRO_COMPAT_NAME="unknown"
-                ;;
-            esac
+          DISTRO="unknown"
+          DISTRO_COMPAT_NAME="unknown"
+          SYSVERSION="unknown"
+          SYSCODENAME="unknown"
         fi
+      else
+        warning "Distribution auto-detection overridden by user. This is not guaranteed to work, and is not officially supported."
+      fi
 
-        case "${DISTRO_COMPAT_NAME}" in
-          centos|ol)
-            SYSVERSION=$(echo "$SYSVERSION" | cut -d'.' -f1)
-            ;;
-        esac
+      supported_compat_names="debian ubuntu centos fedora opensuse ol arch"
+
+      if str_in_list "${DISTRO}" "${supported_compat_names}"; then
+          DISTRO_COMPAT_NAME="${DISTRO}"
       else
-        DISTRO="unknown"
-        DISTRO_COMPAT_NAME="unknown"
-        SYSVERSION="unknown"
-        SYSCODENAME="unknown"
-        SYSARCH="$(uname -m)"
+          case "${DISTRO}" in
+          opensuse-leap)
+              DISTRO_COMPAT_NAME="opensuse"
+              ;;
+          cloudlinux|almalinux|rocky|rhel)
```

Comment:
Are we missing Oracle Linux here?

Reply:
This particular case just handles aliasing of distribution IDs. Oracle Linux is handled by the previous case in the containing `if` block, because we support it natively instead of needing to alias it to a different distro ID.

Reply:
Thanks!